firmware/main.c

```diff
@@ -93,6 +94,7 @@ int main(void)
     
     // MODE SELECT. Only one can be uncommented. TODO: Implement a config.h
```

Comment:
how resource constrained are we right now? Can both fit in?
Alternatively you could make it always start in bitbang and go to sub-modes on request. Although this will possibly interfere with more advanced command handling later I guess

Reply:
We can fit both modes in and have a menu in main.c to select which mode to use. I think that would be simpler than adding routines in the programming modes to switch. Although that could be done with a reset.

Shall I add the main.c menu for the next PR?

Reply:
Possibly...its probably fine for short term. long term we might throw away the parsers entirely to do raw USB. In that case there will be a bit of a conflict on the USB interface I think